Templates/VSIX/Project Templates/cpp-winui-template/WinUI3TemplateCpp/MainWindow.xaml.cpp

```diff
@@ -40,63 +41,49 @@ namespace winrt::$safeprojectname$::implementation
+            GetSystemMetrics(SM_CXICON), GetSystemMetrics(SM_CYICON),
             LR_LOADFROMFILE | LR_SHARED);
-        SendMessage(hwnd, WM_SETICON, ICON_BIG, (LPARAM)hBigIcon);
+        SendMessageW(hwnd, WM_SETICON, ICON_BIG, reinterpret_cast<LPARAM>(hBigIcon));
     }
 
-    void MainWindow::SetWindowSize(const HWND hwnd, const int width, const int height)
+    void MainWindow::SetWindowSize(HWND hwnd, int width, int height)
     {
         // Win32 uses pixels and WinUI 3 uses effective pixels, so you should apply the DPI scale factor
-        UINT dpi = GetDpiForWindow(hwnd);
-        float scalingFactor = static_cast<float>(dpi) / 96;
-        int widthScaled = static_cast<int>(width * scalingFactor);
-        int heightScaled = static_cast<int>(height * scalingFactor);
+        const UINT dpi = GetDpiForWindow(hwnd);
+        const float scalingFactor = static_cast<float>(dpi) / 96;
+        const int widthScaled = static_cast<int>(width * scalingFactor);
+        const int heightScaled = static_cast<int>(height * scalingFactor);
 
         SetWindowPos(hwnd, nullptr, 0, 0, widthScaled, heightScaled, SWP_NOMOVE | SWP_NOZORDER);
     }
 
     void MainWindow::PlacementCenterWindowInMonitorWin32(HWND hwnd)
     {
-        RECT rc;
-        GetWindowRect(hwnd, &rc);
-        ClipOrCenterRectToMonitorWin32(&rc);
-        SetWindowPos(hwnd, nullptr,
-            rc.left, rc.top, 0, 0,
-            SWP_NOSIZE |
-            SWP_NOZORDER |
-            SWP_NOACTIVATE);
+        RECT windowMontiorRectToAdjust;
+        GetWindowRect(hwnd, &windowMontiorRectToAdjust);
+        ClipOrCenterRectToMonitorWin32(windowMontiorRectToAdjust);
+        SetWindowPos(hwnd, nullptr, windowMontiorRectToAdjust.left,
+            windowMontiorRectToAdjust.top, 0, 0, SWP_NOSIZE | SWP_NOZORDER | SWP_NOACTIVATE);
     }
 
-    void MainWindow::ClipOrCenterRectToMonitorWin32(RECT* prc)
+    void MainWindow::ClipOrCenterRectToMonitorWin32(RECT& prc)
     {
-        HMONITOR hMonitor;
-        RECT rc;
-        int w = prc->right - prc->left;
-        int h = prc->bottom - prc->top;
-
-        hMonitor = MonitorFromRect(prc, MONITOR_DEFAULTTONEAREST);
-        MONITORINFO mi;
-        mi.cbSize = sizeof(MONITORINFO);
+        MONITORINFO mi{ sizeof(mi) };
+        GetMonitorInfoW(MonitorFromRect(&prc, MONITOR_DEFAULTTONEAREST), &mi);
 
-        GetMonitorInfo(hMonitor, &mi);
+        const auto& rcWork = mi.rcWork;
+        const int w = (&prc)->right - (&prc)->left;
+        const int h = (&prc)->bottom - (&prc)->top;
 
-        rc = mi.rcWork;
-        prc->left = rc.left + (rc.right - rc.left - w) / 2;
-        prc->top = rc.top + (rc.bottom - rc.top - h) / 2;
-        prc->right = prc->left + w;
-        prc->bottom = prc->top + h;
+        (&prc)->left = rcWork.left + (rcWork.right - rcWork.left - w) / 2;
+        (&prc)->top = rcWork.top + (rcWork.bottom - rcWork.top - h) / 2;
+        (&prc)->right = (&prc)->left + w;
```

Comment:
These are more naturally written as just `prc.right = prc.left + w;` - you don't need to take the address of the reference to use `operator->` on them.  Except for the call to `MonitorFromRect` which does need `RECT*` as the parameter type.

(Also, consider renaming `prc` to just `rc`, or `rect`)

Reply:
Will rename to `rc`. For context, these are the docs that the code is copied from: https://docs.microsoft.com/en-us/windows/win32/gdi/positioning-objects-on-a-multiple-display-setup